### ISSUE TITLE
Cannot compute `sizeof` generic type in alias

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/Nim#24045]
### Description

```nim
type ArrayBuf*[N: static int, T] = object
  buf: array[N, T]

func maxLen(T: type): int =
  sizeof(T) * 2

type MyBuf[I: type] = ArrayBuf[maxLen(I), byte]

var v: MyBuf[int]
```

note `I: type` in alias - removing it gives #24043 instead

### Nim Version

2.0.8

### Current Output

```text
testit.nim(5, 9) Error: 'sizeof' requires '.importc' types to be '.completeStruct'
```

**Comment by metagn** [comment]
Would guess fixed by #23983 but not sure

Probably shouldn't backport but add test case instead

**Comment by ringabout** [comment]
!nim c
```nim
let code = """
type ArrayBuf*[N: static int, T] = object
  buf: array[N, T]

func maxLen(T: type): int =
  sizeof(T) * 2

type MyBuf[I: type] = ArrayBuf[maxLen(I), byte]

var v: MyBuf[int]
"""

# foo()
writeFile("test.nim", code)

import std/osproc

let (_, exitCode) = execCmdEx("nim c -r test.nim")
doAssert exitCode != 0
```
